compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/ic/IrFactoryImplForWasmIC.kt

```diff
@@ -47,19 +47,9 @@ class IrFactoryImplForWasmIC(stageController: StageController) : AbstractIrFacto
         return this
     }
 
-    override fun declarationSignature(declaration: IrDeclaration): IdSignature? {
-        when (declaration) {
```

Comment:
This was blocking the register of declarations restricted to `IrEnumEntry`. There may be more. Just to avoid dealing with problems and then finding out it was related to this filter, just removed it for now, and will add it later once everything works.

Reply:
So why not just to add `IrEnumEntry`?

Reply:
It was done at the beginning when no tests were passing. Sure I can put it back now and see if it causes any more errors.

Reply:
Returning it with `IrEnumEntry` added fails 3 additional tests. Maybe add it later?

Reply:
Fixed. Needed `IrValueParameter` to be included as well.

Reply:
Hmm, we are really have a `restrictednTo` the value parameter? Interesting...

Reply:
I didn't search too much about this, just saw what declarations are calling this function, and by trial and error found that `IrValueParameter` is needed for 3 tests to pass.